sql/2021/mobile-web/client_hints.sql

```diff
@@ -0,0 +1,64 @@
+# standardSQL
+# Usage of client hint directives
+CREATE TEMPORARY FUNCTION getClientHints(headers STRING)
+RETURNS ARRAY<STRING> LANGUAGE js AS """
+try {
+  const header_name = 'Accept-CH';
+  const parsed_headers = JSON.parse(headers);
+  const matching_headers = parsed_headers.filter(h => h.name.toLowerCase() == header_name.toLowerCase());
+  if (matching_headers.length <= 0) {
+    return [];
+  }
+
+  const unique_directives = new Set();
+  for (const header of matching_headers) {
+    const directives = header.value.split(/\\s*,\\s*/);
+    for (const directive of directives) {
+      unique_directives.add(directive);
+    }
+  }
+
+  return Array.from(unique_directives);
+} catch (e) {
+  return [];
+}
+""";
+
+SELECT
+  client,
+  total_sites,
+  SUM(COUNT(DISTINCT page)) OVER (PARTITION BY client) AS total_sites_using_ch,
```

Comment:
As you are using `HAVING` clause this short cut to get the total sites will not be correct. You may have to do a `JOIN` here instead.

Reply:
Huh, yeah you're right. Removed the `HAVING` and got slightly different results.

Removing the `HAVING` from the query